plugins/inputs/elasticsearch_query/elasticsearch_query.go

```diff
@@ -0,0 +1,287 @@
+  date_field = "@timestamp" 		# Timestamp field, mandatory
+  query_period = "1m"  				# Time window to query (eg. "1m" to query documents from last minute). Normally should be set to same as collection interval
+ 
+# Search the maximum response time per method and per URI
+[[inputs.elasticsearch_query.aggregation]]
+  measurement_name = "http_logs"
+  index = "my-index-*"
+  filter_query = "*"
+  metric_fields = ["response_time"]
+  metric_function = "max"
+  tags = ["method.keyword","URI.keyword"]
+  include_missing_tag = false
+  missing_tag_value = "null"
+  date_field = "@timestamp"
+  query_period = "1m"
+
+# Search number of documents matching a filter query
+[[inputs.elasticsearch_query.aggregation]]
+  measurement_name = "http_logs"
+  index = "*"
+  filter_query = "product_1 AND HEAD"
+  query_period = "1m"
+  date_field = "@timestamp"
+
+# Search number of documents matching a filter query per response status code
+[[inputs.elasticsearch_query.aggregation]]
+  measurement_name = "http_logs"
+  index = "*"
+  filter_query = "downloads"
+  tags = ["response.keyword"]
+  include_missing_tag = false
+  date_field = "@timestamp"
+  query_period = "1m"
+`
+
+type ElasticsearchQuery struct {
+	URLs                []string `toml:"urls"`
+	Username            string
+	Password            string
+	EnableSniffer       bool
+	Tracelog            bool
+	Timeout             internal.Duration
+	HealthCheckInterval internal.Duration
+	Aggregations        []Aggregation `toml:"aggregation"`
+	SSLCA               string        `toml:"ssl_ca"`   // Path to CA file
+	SSLCert             string        `toml:"ssl_cert"` // Path to host cert file
+	SSLKey              string        `toml:"ssl_key"`  // Path to cert key file
+	InsecureSkipVerify  bool          // Use SSL but skip chain & host verification
+	Client              *elastic.Client
+	acc                 telegraf.Accumulator
+}
+
+type Aggregation struct {
+	Index             string
+	MeasurementName   string
+	FilterQuery       string
+	QueryPeriod       internal.Duration
+	MetricFields      []string `toml:"metric_fields"`
+	DateField         string
+	Tags              []string `toml:"tags"`
+	IncludeMissingTag bool
+	MissingTagValue   string
+	MetricFunction    string
+}
+
+type aggKey struct {
+	measurement string
+	name        string
+	function    string
+	field       string
+}
+
+type aggregationQueryData struct {
+	aggKey
+	isParent    bool
+	aggregation elastic.Aggregation
+}
+
+func (e *ElasticsearchQuery) init() error {
+	if e.URLs == nil {
+		return fmt.Errorf("Elasticsearch urls is not defined")
+	}
+
+	err := e.connectToES()
+	return err
+}
+
+func (e *ElasticsearchQuery) connectToES() error {
+
+	var clientOptions []elastic.ClientOptionFunc
+
+	tlsCfg, err := internal.GetTLSConfig(e.SSLCert, e.SSLKey, e.SSLCA, e.InsecureSkipVerify)
+	if err != nil {
+		return err
+	}
+	tr := &http.Transport{
```

Comment:
Make the transport and http.Client a member of `ElasticsearchQuery`, again similar to [the apache plugin](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/apache/apache.go#L69)